build_container/build_container_ubuntu.sh

```diff
@@ -40,27 +40,46 @@ esac
 apt-get update -y
 
 apt-get install -y --no-install-recommends docker-ce-cli wget make cmake git python python-pip python-setuptools python3 python3-pip \
-  python3-setuptools unzip bc libtool ninja-build automake zip time gdb strace tshark tcpdump patch xz-utils rsync ssh-client google-cloud-sdk \
+  python3-setuptools unzip bc libtool automake zip time gdb strace tshark tcpdump patch xz-utils rsync ssh-client google-cloud-sdk \
   libncurses-dev doxygen graphviz
 
 # Python 3.8
 add-apt-repository -y ppa:deadsnakes/ppa
 apt-get update
 apt install -y python3.8
 
-LLVM_VERSION=9.0.0
+# Install ninja-build 1.8.2 from binary
 case $ARCH in
     'ppc64le' )
+        ninja_deb="ninja-build_1.8.2-1_ppc64el.deb"
+        ;;
+    'x86_64' )
+        ninja_deb="ninja-build_1.8.2-1_amd64.deb"
+        ;;
+    'aarch64' )
+        ninja_deb="ninja-build_1.8.2-1_arm64.deb"
+        ;;
+esac
+wget https://launchpad.net/ubuntu/+archive/primary/+files/${ninja_deb}
+dpkg -i ${ninja_deb}
+rm ${ninja_deb}
+
+# Set LLVM version for each cpu architecture.
+case $ARCH in
+    'ppc64le' )
+        LLVM_VERSION=9.0.0
         LLVM_DISTRO=powerpc64le-linux-ubuntu-16.04
         LLVM_SHA256SUM=a8e7dc00e9eac47ea769eb1f5145e1e28f0610289f07f3275021f0556c169ddf
         ;;
     'x86_64' )
+        LLVM_VERSION=9.0.0
         LLVM_DISTRO=x86_64-linux-gnu-ubuntu-16.04
         LLVM_SHA256SUM=5c1473c2611e1eac4ed1aeea5544eac5e9d266f40c5623bbaeb1c6555815a27d
         ;;
     'aarch64' )
+        LLVM_VERSION=8.0.0
```

Comment:
Add a comment why this is 8.0.0?

Reply:
Hi, Lizan, 
Actually, When i am using clang 9.0.0 to build envoy test, i encounter some problems about envoy test cases.
```
FAIL: //test/extensions/health_checkers/redis:config_test (see /build/tmp/_bazel_bazel/b570b5ccd0454dc9af9f65ab1833764d/execroot/envoy/bazel-out/aarch64-fastbuild/testlogs/test/extensions/health_checkers/redis/config_test/test.log)
INFO: From Testing //test/extensions/health_checkers/redis:config_test:
==================== Test output for //test/extensions/health_checkers/redis:config_test:
WARNING: Perftools heap leak checker is active -- Performance may suffer
[==========] Running 6 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 6 tests from HealthCheckerFactoryTest
[ RUN      ] HealthCheckerFactoryTest.CreateRedisDeprecated
[       OK ] HealthCheckerFactoryTest.CreateRedisDeprecated (671 ms)
[ RUN      ] HealthCheckerFactoryTest.CreateRedis
[       OK ] HealthCheckerFactoryTest.CreateRedis (160 ms)
[ RUN      ] HealthCheckerFactoryTest.CreateRedisWithoutKeyDeprecated
[       OK ] HealthCheckerFactoryTest.CreateRedisWithoutKeyDeprecated (154 ms)
[ RUN      ] HealthCheckerFactoryTest.CreateRedisWithoutKey
[       OK ] HealthCheckerFactoryTest.CreateRedisWithoutKey (158 ms)
[ RUN      ] HealthCheckerFactoryTest.CreateRedisWithLogHCFailure
[       OK ] HealthCheckerFactoryTest.CreateRedisWithLogHCFailure (160 ms)
[ RUN      ] HealthCheckerFactoryTest.CreateRedisViaUpstreamHealthCheckerFactory
[       OK ] HealthCheckerFactoryTest.CreateRedisViaUpstreamHealthCheckerFactory (153 ms)
[----------] 6 tests from HealthCheckerFactoryTest (1456 ms total)

[----------] Global test environment tear-down
[==========] 6 tests from 1 test suite ran. (1456 ms total)
[  PASSED  ] 6 tests.
external/bazel_tools/tools/test/test-setup.sh: line 310: 16324 Segmentation fault      (core dumped) "${TEST_PATH}" "$@" 2>&1
```
Obviously, the result of test case has been successfully, but it finally failed. I check the environments carefully, but not find out root cause.
At the same time, i change to another version of clang, including clang 9.0.1 and 8.0.0. Both of them are OK. So i choose to temporary downgrade arm64 clang to 8.0.0 since there is no official release about 9.0.1 and 10.0.0.
Until there will be official release ( 9.0.1 / 10.0.0 ) about clang, it will be upgraded to later version.
Is it feasible ?

Reply:
I'm not against using 8.0.0, just add a comment to this script.

Reply:
Thanks for your comments. Has updated.